### PR TITLE
For #20596 remove perf.startup probes

### LINKED ISSUE #20596: Remove StartupTimeline & start up subsection duration probes
While reviewing telemetry probes for renewal https://github.com/mozilla-mobile/fenix/issues/20520, we decided not to renew:
```
Startup.timeline:
  framework_primary
  framework_secondary
  framework_start_error
  framework_start_read_error
  clock_ticks_per_second_v2

perf.startup:
 app_on_create_to_glean_init
 app_on_create_to_megazord_init
 app_on_create_to_setup_in_main
 home_activity_on_create
 home_activity_on_start
 home_fragment_on_create_view
 home_fragment_on_view_created
 base_bfragment_on_create_view
 base_bfragment_on_view_created
```

Because we're not actively using them and have no intention to do so in the near future. We should remove this code.

### PR DESCRIPTION
**Pull request by RotBolt**
### Pull Request checklist

- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [X] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Closes #20596 

I have removed `perf.startup probes` and its usage from the application code. Also removed the associated functions and files which are no longer used after removal of `perf.startup probes`

### CODE REVIEW

**Comment by RotBolt** [comment]
hello @mcomella  as discussed I have first raised the PR for removal for `perf.startup probes`.
Once this removal is approved. I will remove the `startup_timeline probes`  in the same PR

File: app/src/main/java/org/mozilla/fenix/ext/Glean.kt
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/ext/Glean.kt:19]: Should we maybe keep this around in case we need it again?

Comment (mcomella) [reply app/src/main/java/org/mozilla/fenix/ext/Glean.kt:19]: Good catch – however, this was actually working around [a bug in Glean](https://bugzilla.mozilla.org/show_bug.cgi?id=1699505) that has since been fixed so the method is unnecessary and we should keep it removed: calling Glean's `measure` directly is the proper solution and we just hadn't updated our code to use the new method yet.

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:240]: nitpick (non-blocking): in your next PR, could you remove this comment? It was actually written for the `measureNoInline` call (iirc, originally the method body was not indented and the line had a weird indentation to cause the linter to be happy but the method body has since been autoformatted such that the comment was irrelevant).

Even though it's more related to this PR than the next one, I don't want to add it to this PR because it'd create add an additional delay that adds overhead for both of us. There's a certain standard of correctness we want to maintain for our project but we need to make deliberate trade-offs about the amount of time required to maintain them. In this case, which commit/PR a comment removal ends up probably isn't going to cause bugs or confusion down the line so it's fine to remove it wherever.

**Review by mcomella — APPROVED** [review]
This looks good to me – thanks for your help, @RotBolt!

> Once this removal is approved. I will remove the startup_timeline probes in the same PR

Let's land this PR and you can open a new one for the additional changes. It's easier to land small PRs when possible because small atomic changes are less work to write and review than larger ones and the faster we land the change, the less likely we are to have merge conflicts.

Now I'll just have to figure out how to merge this without blocking the merge queue again... 😅

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt
Comment (RotBolt) [reply app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:240]: Sure will remove this comment on next PR

**Comment by RotBolt** [comment]
Oh, that's unfortunate. Hope it will get fix sooner.
On the note of making another PR. I actually have the work ready for
`startup.timeline` probe.
So should I still make PR for that ? Also would that PR face the same issue
of CI like this one ?

**Comment by mcomella** [comment]
@RotBolt Sorry if I was misleading – we're able to merge the code but it'll just happen in a duplicate PR and we'll close the original PR. I merged your code in https://github.com/mozilla-mobile/fenix/pull/20790 and I'll close this PR.

To try to clarify, we have restrictions around executing full CI on PRs from folks outside of our GH org. We had a system to do it but we changed our process slightly and the system broke, as we learned from your previous PR. 😁 However, we can workaround it by having someone from our GH org open the PR instead with your code as you wrote it (which was #20790).

> I actually have the work ready for `startup.timeline` probe. So should I still make PR for that ? Also would that PR face the same issue of CI like this one ?

Yes please open a new PR and flag me for review! We'll be able to land your code and it'll go more smoothly now that we know how to work around the bug in our CI merging process.

And again, I'm closing this PR but don't worry – your code was already merged in #20790. You should be able to see it on the `upstream/main` branch if you fetch. :)

**Pull request closed without merging**
